src/transformers/utils/sentencepiece_model_pb2_new.py

```diff
@@ -14,9 +14,14 @@
-)
+pool = _descriptor_pool.Default()
+# Before adding the serialized file, try to find it in the pool (that is global).
+try:
+    DESCRIPTOR = pool.FindFileByName("sentencepiece_model.proto")
```

Comment:
My concern here is: when we get this, does it (always) works with `transformers`? (we might get strange stuff for this object from other libraries).

Reply:
No, we do not have a real guarantee that it will _always_ work. That is why I said it mitigates the issues, it doesn't remove it altogether. Having said that, if some other code loaded before transformers uses protobuf and adds a file whose name matches `sentencepiece_model.proto`, I guess it's not going to be anything that is not sentencepiece! Alternatively I could do it the other way around, I can try to add the fail, and if it raises a `TypeError` I can try to do the `FindFileByName`.
In both cases I could also add a warning if `FindFileByName` is used.

Reply:
No, I don't mean `it's sentencepiece or not sentencepiece`. What I worry a bit is it's (may) not the one we have and we don't know what would happen ...

Reply:
that could probably be controlled by an  environment variable however.

Reply:
the problem is that is the `sentencepiece_model.proto` was added in the current process, there is no way to add it again. We cannot be sure it's the right one, as it does not provide a version or a hash. And if we try to load it, it will raise an exception from the C extension. So that is why I thought that assuming it's the same proto is probably a fair fallback solution.

Reply:
if you have a better option, I'd be happy to consider it though!